jax/_src/numpy/fft.py

```diff
@@ -78,7 +78,18 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
     else:
       s = [a.shape[axis] for axis in axes]
 
-  transformed = lax.fft(a, fft_type, s)
+  # It's faster to compute a real->complex FFT and form the second half by
+  # complex conjugation than to promote to complex and perform a
+  # complex->complex FFT.
```

Comment:
Doing it this way also has implications for autodiff, e.g., due to the fact that `fft`/`ifft` and `rfft`/`irfft` have very different transpose rules.

So I wonder: would be better to put this inside the `lax.fft` primitive instead?

At the very least, it would be good to make sure that this short-cut doesn't slow down transposes.